modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/type/EmailNotificationType.java

```diff
@@ -225,6 +226,18 @@ public void sendNotification(NotificationContext notificationContext)
 						notificationRecipientSetting.getValueMap(),
 						notificationContext);
 				}
+			).put(
+				"singleRecipient",
+				() -> {
+					NotificationRecipientSetting notificationRecipientSetting =
+						notificationRecipientSettingLocalService.
+							getNotificationRecipientSetting(
+								notificationRecipient.
+									getNotificationRecipientId(),
+								"singleRecipient");
+
+					return notificationRecipientSetting.getValue();
```

Comment:
Yeah, we need this property because it will be saved in the database. I understood that the intention of the `evaluatedNotificationRecipientSettings` map is evaluate all the properties to be added to the `notificationContext` in the `prepareNotificationContext` method.
We need to evaluate this in the map because after the `prepareNotificationContext` execution, we add a NotificationQueueEntry based in the notificationContext.

```
notificationQueueEntryLocalService.addNotificationQueueEntry(
					notificationContext);
```

The NotificationRecipientSettings are updated [here](https://github.com/liferay/liferay-portal/blob/7f9fd3cb9b0200ccdb18ef90af558f0156844c69/modules/apps/notification/notification-service/src/main/java/com/liferay/notification/service/impl/NotificationQueueEntryLocalServiceImpl.java#L105-L116).

Reply:
I will mark this as resolved, but feel free to open again if I'm forgetting something

Reply:
Agreed